research/gaia/gremlin/gremlin_core/src/structure/graph.rs

```diff
@@ -45,73 +45,85 @@ impl<E: Element + Send + Sync> FromPb<Option<pb::QueryParams>> for QueryParams<E
     where
         Self: Sized,
     {
-        let mut query_params = QueryParams::default();
-        if let Some(query_params_pb) = query_params_pb {
-            if let Some(labels) = query_params_pb.labels {
-                query_params.labels =
-                    labels.labels.into_iter().map(|l| Label::Id(l as LabelId)).collect();
-            }
-            if let Some(limit) = query_params_pb.limit {
-                query_params.limit = Some(limit.limit as usize);
-            }
-            if let Some(required_properties) = query_params_pb.required_properties {
-                query_params.set_props(required_properties);
-            }
-            if let Some(ref predicates) = query_params_pb.predicates {
-                if let Some(filter) = pb_chain_to_filter(predicates)? {
-                    query_params.set_filter(filter);
-                }
-            }
-            if let Some(extra_params) = query_params_pb.extra_params {
-                query_params.set_extra_params(extra_params)
-            }
-        }
-        Ok(query_params)
+        query_params_pb.map_or(Ok(QueryParams::default()), |query_params_pb| {
+            QueryParams::default()
+                .with_labels(query_params_pb.labels)?
+                .with_filter(query_params_pb.predicates)?
+                .with_limit(query_params_pb.limit)?
+                .with_props(query_params_pb.required_properties)?
```

Comment:
I'd like the name of `with_required_properties`.

Reply:
Done